File: FE/src/features/user/services/userService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type {
  CreateUserRequest,
  UpdateUserRequest,
  UserResponse,
} from '../types/user.types'

export const userService = {
  getAll: (page = 1, size = 10, sort = 'id,asc', active?: boolean) => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    if (active !== undefined) {
      params.set('active', String(active))
    }
    return apiClient<PaginatedResult<UserResponse>>(`/users?${params}`)
  },

  getById: (id: number) => apiClient<UserResponse>(`/users/${id}`),

  create: (data: CreateUserRequest) =>
    apiClient<UserResponse>('/users', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  update: (data: UpdateUserRequest) =>
    apiClient<UserResponse>('/users', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  disable: (id: number) =>
    apiClient<UserResponse>(`/users/${id}`, { method: 'DELETE' }),

  enable: (id: number) =>
    apiClient<UserResponse>(`/users/${id}/enable`, { method: 'POST' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/user/UserService.java
package Manager_vnd.Manager.feature.user;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.user.dto.CreateUserRequest;
import Manager_vnd.Manager.feature.user.dto.UpdateUserRequest;
import Manager_vnd.Manager.feature.user.dto.UserResponse;

public interface UserService {

    PaginatedResult<UserResponse> getAllUsers(int page, int size, String sort, Boolean active);

    UserResponse getUserById(long id);

    UserResponse createUser(CreateUserRequest request);

    UserResponse updateUser(UpdateUserRequest request);

    UserResponse disableUser(long id);

    UserResponse enableUser(long id);
}
